onnx2torch/node_converters/scatter_nd.py

```diff
@@ -19,13 +18,17 @@ def forward(self, data: torch.Tensor, indices: torch.Tensor, updates: torch.Tens
         # There is no scatter nd for torch, use following formula:
         # https://github.com/onnx/onnx/blob/master/docs/Operators.md#ScatterND
         output = data.clone()
-        update_indices = indices.shape[:-1]
-        for idx in np.ndindex(update_indices):
-            output[tuple(indices[idx])] = updates[idx]
 
         if torch.onnx.is_in_onnx_export():
             return _ScatterNDExportToOnnx.set_output_and_apply(output, data, indices, updates)
 
+        ind_dim = indices.dim()
+        # last dimension is a partial-index into data
+        indices = indices.reshape((-1, indices.shape[-1])).T.tolist()
```

Comment:
Transpose can be skipped
```python
indices = indices.reshape((indices.shape[-1], -1)).tolist()
```

Reply:
indices.reshape((indices.shape[-1], -1)) != indices.reshape((-1, indices.shape[-1])).T

We need not only needed shape, but and swapaxes. Here 'indices.reshape((indices.shape[-1], -1))' we just unroll last dim, but 'indices.reshape((-1, indices.shape[-1])).T' here we have tensor with indices for each dim in data.